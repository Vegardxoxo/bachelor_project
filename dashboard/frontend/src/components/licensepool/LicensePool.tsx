import {Box, Grid, Stack, Typography} from '@mui/material';
import React, {useEffect, useState} from 'react';
import SoftwareSearchBar from '../search/SoftwareSeachBar';
import PoolTable from "./PoolTable/PoolTable";
import {LicensePoolData} from "../../Interfaces";
import {fetchPoolData, fetchSoftwareUsedInOrg} from "../../api/calls";
import Checkbox from '@mui/material/Checkbox';
import FormControlLabel from '@mui/material/FormControlLabel';
import Pagination from "@mui/material/Pagination";
import ActiveLastBreadcrumb from "../dashboard/ActivateLastBreadcrumb/ActivateLastBreadcrumb";
import {useRecoilValue} from "recoil";
import {refreshTableAtom} from "../../globalVariables/variables";

function LicensePool() {
    const org = JSON.parse(localStorage.getItem('organization') ?? 'null');
    const [searchTerm, setSearchTerm] = useState<string>();
    const [orgSoftware, setOrgSoftware] = useState<string[]>([]);
    const [data, setData] = useState<LicensePoolData[]>([]);
    const [checked, setChecked] = useState<boolean>(false);
    const [errorMessage, setErrorMessage] = useState<string>('');
    const [currentPage, setCurrentPage] = React.useState(1);
    const ITEMS_PER_PAGE = 10;
    const [count, setCount] = useState<number>(0);
    const [sortBy, setSortBy] = useState<string>('application_name')
    const refreshTable = useRecoilValue(refreshTableAtom)

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setChecked(event.target.checked);
        setCurrentPage(1);
    };

    const handlePageChange = async (event: React.ChangeEvent<unknown>, value: number) => {
        setCurrentPage(value);
    };

    const handleSorting = async (sortBy: string) => {
        setSortBy(sortBy);
        await fetchData()
    };

    // Function that gets input from the searchBar component.
    const updateSearchTerm = (term: string) => {
        setSearchTerm(term);
        setCurrentPage(1);
    }

    useEffect(() => {
        setErrorMessage('')
        // Fetches distinct software names.
        const fetchSoftwareNames = async () => {
            try {
                const data: string[] | undefined = await fetchSoftwareUsedInOrg('available', 'true', checked ? org as string : '');
                if (data !== undefined) {
                    setOrgSoftware(data);
                }
            } catch (error) {
                console.error('Error fetching software names:', error);
            }
        };
        fetchSoftwareNames();
    }, [checked, refreshTable]);

    const fetchData = async () => {
        try {
            const {
                results,
                count,
                error,
                message
            } = await fetchPoolData(currentPage, sortBy as string,
                searchTerm, checked ? org as string : undefined);
            if (error) {
                setErrorMessage(message);
                setData([]);
            } else {
                setData(results);
                setCount(count);
            }
        } catch (error) {
            console.error('Error fetching license data:', error);
        }
    };

    useEffect(() => {
        fetchData();
    }, [currentPage, sortBy, checked, searchTerm, refreshTable]);


    return (
        <div>
            <Grid>
                <ActiveLastBreadcrumb/>
            </Grid>
            <Box id={'licensepool_container'}
                 style={{display: 'flex', justifyContent: 'center', alignContent: "center", marginTop: "20px"}}>
                <Grid container className='license_pool' justifyContent={"center"}>
                    <Grid container justifyContent="center" alignItems="center" className={'license_table'}
                          width={"75%"}>
                        <Stack direction={"column"} width={"95%"} marginBottom={"10px"} marginLeft={'-1%'}>
                            <h2 style={{
                                fontFamily: 'Source Sans Pro, sans-serif',
                                fontSize: '30pt',
                                marginTop: -0.6,
                                fontWeight: 400
                            }}>
                                Lisensportalen
                            </h2>
                            <h4 style={{
                                fontFamily: 'Source Sans Pro, sans-serif',
                                fontStyle: "italic",
                                fontWeight: 200,
                                marginTop: "-2%"
                            }}>
                                -Velg gjenbruk!
                            </h4>
                        </Stack>
                        <Stack direction={'row'} spacing={5} width={"95%"} marginBottom={"30px"} alignItems="center"
                               marginTop={"10px"}>
                            <SoftwareSearchBar
                                data-testid={'software_search_bar'}
                                setSelectedSoftware={updateSearchTerm}
                                data={orgSoftware}/>
                            <FormControlLabel
                                control={
                                    <Checkbox
                                        checked={checked}
                                        onChange={handleChange}
                                        inputProps={{'aria-label': 'controlled'}}
                                    />
                                }
                                label={<Typography style={{fontFamily: 'Source Sans Pro,sans-serif', fontSize: '13pt'}}>Vis
                                    bare lisenser fra min enhet</Typography>}
                            />
                            {errorMessage && <h3 style={{color: 'red'}}>{errorMessage}</h3>}

                        </Stack>
                        <Stack direction={'column'} width={"100%"} marginLeft={3}>
                            <PoolTable data={data} handleSorting={handleSorting}/>
                            <Pagination
                                count={Math.ceil(count / ITEMS_PER_PAGE)}
                                page={currentPage}
                                onChange={handlePageChange}
                                color={"primary"}
                                style={{marginTop: '1rem'}}
                            />

                        </Stack>

                    </Grid>
                </Grid>
            </Box>
        </div>
    );


}

export default LicensePool;