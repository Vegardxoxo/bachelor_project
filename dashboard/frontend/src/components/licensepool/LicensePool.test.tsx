import React from 'react';
import {cleanup, render} from '@testing-library/react';
import {RecoilRoot} from 'recoil';
import LicensePool from "./LicensePool";
import {MemoryRouter} from 'react-router-dom';
import renderer from "react-test-renderer";
import '@testing-library/jest-dom';
import 'isomorphic-fetch';
import {act} from 'react-dom/test-utils';

const data = [
    {
        "application_name": "APSIS Pro [Web]",
        "freed_by_organization": "IT-tjenesten",
        "details": [
            {
                "id": 71,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "APSIS Pro [Web]",
                "date_added": "2023-05-02",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": null,
                "spc_id": 147822
            }
        ]
    },
    {
        "application_name": "Anaconda Python 2022",
        "freed_by_organization": "IT-tjenesten",
        "details": [
            {
                "id": 66,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "Anaconda Python 2022",
                "date_added": "2023-04-26",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": null,
                "spc_id": 147830
            }
        ]
    },
    {
        "application_name": "EasyChair [Web]",
        "freed_by_organization": "IT-tjenesten",
        "details": [
            {
                "id": 69,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "EasyChair [Web]",
                "date_added": "2023-05-02",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": null,
                "spc_id": 147828
            }
        ]
    },
    {
        "application_name": "Microsoft 365 Family",
        "freed_by_organization": "IT-tjenesten",
        "details": [
            {
                "id": 70,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "Microsoft 365 Family",
                "date_added": "2023-05-02",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": null,
                "spc_id": 147836
            }
        ]
    },
    {
        "application_name": "Microsoft Office 2016 Access",
        "freed_by_organization": "IT-tjenesten",
        "details": [
            {
                "id": 67,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "Microsoft Office 2016 Access",
                "date_added": "2023-04-26",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": 990,
                "spc_id": 95746
            },
            {
                "id": 75,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "Microsoft Office 2016 Access",
                "date_added": "2023-05-03",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": null,
                "spc_id": 147831
            }
        ]
    },
    {
        "application_name": "Microsoft Office 2016 Excel",
        "freed_by_organization": "IT-tjenesten",
        "details": [
            {
                "id": 76,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "Microsoft Office 2016 Excel",
                "date_added": "2023-05-03",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": null,
                "spc_id": 147818
            }
        ]
    },
    {
        "application_name": "Microsoft Office 2016 OneNote",
        "freed_by_organization": "IT-tjenesten",
        "details": [
            {
                "id": 77,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "Microsoft Office 2016 OneNote",
                "date_added": "2023-05-03",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": null,
                "spc_id": 147815
            }
        ]
    },
    {
        "application_name": "Microsoft Office 2016 Outlook",
        "freed_by_organization": "IT-tjenesten",
        "details": [
            {
                "id": 78,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "Microsoft Office 2016 Outlook",
                "date_added": "2023-05-03",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": null,
                "spc_id": 147835
            }
        ]
    },
    {
        "application_name": "Microsoft Office 2016 PowerPoint",
        "freed_by_organization": "IT-tjenesten",
        "details": [
            {
                "id": 72,
                "freed_by_organization": "IT-tjenesten",
                "application_name": "Microsoft Office 2016 PowerPoint",
                "date_added": "2023-05-02",
                "family": null,
                "family_version": null,
                "family_edition": null,
                "price": null,
                "spc_id": 147820
            }
        ]
    }
]
const count = 13
const softwareNames = [
    "APSIS Pro [Web]",
    "Anaconda Python 2022",
    "EasyChair [Web]",
    "Microsoft 365 Family",
    "Microsoft Office 2016 Access",
    "Microsoft Office 2016 Excel",
    "Microsoft Office 2016 OneNote",
    "Microsoft Office 2016 Outlook",
    "Microsoft Office 2016 PowerPoint",
    "Microsoft Office 2016 Publisher",
    "Microsoft Office 2016 Word",
    "SecMaker Net iD 6"
]
describe('LicensePool', () => {
    beforeEach(() => {
        jest.spyOn(global, 'fetch').mockImplementation((url) => {
            const urlString = typeof url === 'string' ? url : url.toString();

            if (urlString.startsWith("http://127.0.0.1:8000/api/pool/get/")) {
                return Promise.resolve(
                    new Response(JSON.stringify({results: data, count}))
                );
            } else if (urlString.startsWith("http://127.0.0.1:8000/api/licenses/software/")) {
                return Promise.resolve(
                    new Response(
                        JSON.stringify(softwareNames)
                    )
                );
            } else {
                return Promise.reject(new Error("Invalid URL"));
            }
        });
    });


    afterEach(() => {
        cleanup();
        jest.clearAllMocks();
    })
    it('renders without crashing', async () => {
        await act(async () => {
            render(
                <MemoryRouter>
                    <RecoilRoot>
                        <LicensePool/>
                    </RecoilRoot>
                </MemoryRouter>
            );
        });
    });

    it('matches snapshot', async () => {
        const testRenderer = renderer.create(
            <MemoryRouter>
                <RecoilRoot>
                    <LicensePool/>
                </RecoilRoot>
            </MemoryRouter>
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    })

});