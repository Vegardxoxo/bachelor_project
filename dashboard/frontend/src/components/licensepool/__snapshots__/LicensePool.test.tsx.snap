// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LicensePool matches snapshot 1`] = `
<div>
  <div
    className="MuiGrid-root css-vj1n65-MuiGrid-root"
  >
    <nav
      aria-label="breadcrumb"
      className="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      id="breadcrumbs"
    >
      <ol
        className="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
      >
        <li
          className="MuiBreadcrumbs-li"
        >
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-le90w3-MuiTypography-root-MuiLink-root"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
          >
            Dashboard
          </a>
        </li>
      </ol>
    </nav>
  </div>
  <div
    className="MuiBox-root css-0"
    id="licensepool_container"
    style={
      Object {
        "alignContent": "center",
        "display": "flex",
        "justifyContent": "center",
        "marginTop": "20px",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container license_pool css-1lym95h-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container license_table css-204e7f-MuiGrid-root"
      >
        <div
          className="css-17qp04y-MuiStack-root"
        >
          <h2
            style={
              Object {
                "fontFamily": "Source Sans Pro, sans-serif",
                "fontSize": "30pt",
                "fontWeight": 400,
                "marginTop": -0.6,
              }
            }
          >
            Lisensportalen
          </h2>
          <h4
            style={
              Object {
                "fontFamily": "Source Sans Pro, sans-serif",
                "fontStyle": "italic",
                "fontWeight": 200,
                "marginTop": "-2%",
              }
            }
          >
            -Velg gjenbruk!
          </h4>
        </div>
        <div
          className="css-sud8pg-MuiStack-root"
        >
          <div
            aria-owns={null}
            className="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1ih8noe-MuiAutocomplete-root"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
          >
            <div
              className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              data-testid="autocomplete-search"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1l2vu2a-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink={true}
                htmlFor="controllable-states-demo"
                id="controllable-states-demo-label"
              >
                Søk
              </label>
              <div
                className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-17yn54g-MuiInputBase-root-MuiOutlinedInput-root"
                onClick={[Function]}
              >
                <input
                  aria-activedescendant={null}
                  aria-autocomplete="list"
                  aria-expanded={false}
                  aria-invalid={false}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoFocus={false}
                  className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled={false}
                  id="controllable-states-demo"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  readOnly={false}
                  required={false}
                  role="combobox"
                  spellCheck="false"
                  type="text"
                  value=""
                />
                <button
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "position": "absolute",
                      "right": 15,
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="SearchIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </button>
                <div
                  className="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                >
                  <button
                    aria-label="Open"
                    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={-1}
                    title="Open"
                    type="button"
                  />
                </div>
                <fieldset
                  aria-hidden={true}
                  className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    className="css-14lo706"
                  >
                    <span>
                      Søk
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                aria-label="controlled"
                checked={false}
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                onChange={[Function]}
                required={false}
                type="checkbox"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              style={
                Object {
                  "fontFamily": "Source Sans Pro,sans-serif",
                  "fontSize": "13pt",
                }
              }
            >
              Vis bare lisenser fra min enhet
            </p>
          </label>
        </div>
        <div
          className="css-ybvm-MuiStack-root"
        >
          <h3>
            Velg programvare 
          </h3>
          <nav
            aria-label="pagination navigation"
            className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
            style={
              Object {
                "marginTop": "1rem",
              }
            }
          >
            <ul
              className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
            >
              <li>
                <button
                  aria-label="Go to previous page"
                  className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                    data-testid="NavigateBeforeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to next page"
                  className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                  disabled={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={-1}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                    data-testid="NavigateNextIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;
